enforcer_synced.go

```diff
@@ -131,8 +131,8 @@ func (e *SyncedEnforcer) LoadIncrementalFilteredPolicy(filter interface{}) error
 
 // SavePolicy saves the current policy (usually after changed with Casbin API) back to file/database.
 func (e *SyncedEnforcer) SavePolicy() error {
-	e.m.RLock()
-	defer e.m.RUnlock()
+	e.m.Lock()
```

Comment:
Will there be a race in SavePolicy, I think it just reads the resource

Reply:
Saving policies cause the adapter to write to, i.e., a database.
A `RemovePolicy()` and the `SavePolicy` will emit concurrent database writes.

Reply:
The adapter itself should ensure transaction security, but it seems that it does not now, it can only be locked here